sky/setup_files/dependencies.py

```diff
@@ -121,7 +121,8 @@
     'cloudflare': aws_dependencies,
     'scp': local_ray,
     'oci': ['oci'] + local_ray,
-    'kubernetes': ['kubernetes>=20.0.0'],
+    # Kubernetes 32.0.0 has an authentication bug: https://github.com/kubernetes-client/python/issues/2333 # pylint: disable=line-too-long
+    'kubernetes': ['kubernetes>=20.0.0,<32.0.0'],
```

Comment:
How about in case there is a new fix later?
```suggestion
    'kubernetes': ['kubernetes>=20.0.0,!=32.*'],
```

Reply:
kubernetes maintainers haven't acknowledged it yet as a bug, so it's unclear if the fix would involve some changes in our code. Might be safer to pin to versions that are known to work